File: 29-chrome-driver/csharp/Chrome.cs
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Remote;
using System;
using System.IO;

public class Chrome
{
    IWebDriver Driver;
    string VendorDirectory = Directory.GetParent(
        Path.GetDirectoryName(typeof(Chrome).Assembly.Location)).
            Parent.FullName + @"\Vendor";

    [SetUp]
    public void SetUp()
    {
        Driver = new ChromeDriver(VendorDirectory);
        //DesiredCapabilities Capabilities = new DesiredCapabilities();
        //DesiredCapabilities.Chrome();
        //Driver = new RemoteWebDriver(new Uri("http://localhost:9515"), Capabilities);
    }

    [TearDown]
    public void TearDown()
    {
        Driver.Quit();
    }

    [Test]
    public void PageLoads()
    {
        Driver.Navigate().GoToUrl("http://the-internet.herokuapp.com");
        Assert.That(Driver.Title.Equals("The Internet"));
    }
}

File: 29-chrome-driver/python/chrome.py
# -*- coding: utf-8 -*-
"""
Implementation of http://elementalselenium.com/tips/29-chrome
"""

import os
import unittest
from selenium import webdriver


class Chrome(unittest.TestCase):

    def setUp(self):
        #chromedriver_path = os.getcwd() + '/vendor/chromedriver'
        #self.driver = webdriver.Chrome(chromedriver_path)
        self.driver = webdriver.Remote('http://localhost:9515', {'browserName': 'chrome'})

    def tearDown(self):
        self.driver.quit()

    def test_example_1(self):
        driver = self.driver
        driver.get('http://the-internet.herokuapp.com/upload')
        assert driver.title == 'The Internet'

if __name__ == "__main__":
    unittest.main()
